core/views.py

```diff
@@ -55,3 +59,45 @@ def search(request, parent_page_id=None):
             'page_type_string': page_type_string,
         })
     )
+
+
+def site_search(request, template=None, results_per_page=10, path=None):
+    # Get default templates
+    if template is None:
+        if hasattr(settings, 'WAGTAILSEARCH_RESULTS_TEMPLATE'):
+            template = settings.WAGTAILSEARCH_RESULTS_TEMPLATE
+        else:
+            template = 'wagtailsearch/search_results.html'
+
```

Comment:
I copied this code from wagtail, so I don't think I want to change it, but I agree it isn't really ideal.

Better yet one could use `getattr` and provide it with a default like this:

`template = getattr(settings, 'WAGTAILSEARCH_RESULTS_TEMPLATE', 'wagtailsearch/search_results.html')`

How much you use exceptions are not is kind of a matter of style, but I do agree most people probably don't use them enough.

But to make maintenance easier I don't think I want to change this.